src/olmo_core/data/utils.py

```diff
@@ -438,10 +438,14 @@ def segment_documents_into_instances(
         rng = get_rng(seed)
         indices = rng.choice(indices.reshape(-1, 2), size=max_instances).reshape(-1)
 
-    with memmap_to_write(target, dtype=indices_dtype, shape=(indices.size,)) as indices_mmap:
-        indices_mmap[:] = indices
+    # NOTE: It's possible to sample 0 instances from small source files. Rather than try to write this empty array we conditionally skip if indices_out is 0.
+    indices_out = len(indices) // 2
 
-    return total_og_docs, len(indices) // 2
+    if indices_out > 0:
+        with memmap_to_write(target, dtype=indices_dtype, shape=(indices.size,)) as indices_mmap:
+            indices_mmap[:] = indices
```

Comment:
Is it going to cause issues elsewhere when we never write to the target memmap file?